nixos/modules/services/network-filesystems/seaweedfs.nix

```
@@ -0,0 +1,597 @@
+
+      rack = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "Current volume server's rack name.";
+      };
+
+      disk = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "[hdd|ssd|<tag>] hard drive or solid state drive or any tag.";
+      };
+
+      idxDir = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = null;
+        description = "Directory to store .idx files.";
+      };
+
+      index = lib.mkOption {
+        type = lib.types.enum [
+          "memory"
+          "leveldb"
+          "leveldbMedium"
+          "leveldbLarge"
+        ];
+        default = "memory";
+        description = "Mode for memory~performance balance.";
+      };
+
+      readMode = lib.mkOption {
+        type = lib.types.enum [
+          "local"
+          "proxy"
+          "redirect"
+        ];
+        default = "proxy";
+        description = "How to deal with non-local volume: not found|proxy to remote node|redirect volume location.";
+      };
+
+      minFreeSpace = lib.mkOption {
+        type = lib.types.str;
+        default = "1";
+        description = "Min free disk space (value<=100 as percentage like 1, other as human readable bytes, like 10GiB).";
+      };
+
+      fileSizeLimitMB = lib.mkOption {
+        type = lib.types.int;
+        default = 256;
+        description = "Limit file size to avoid out of memory.";
+      };
+
+      metricsPort = lib.mkOption {
+        type = lib.types.nullOr lib.types.port;
+        default = null;
+        description = "Prometheus metrics listen port.";
+      };
+
+      whiteList = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        description = "Ip addresses having write permission. No limit if empty.";
+      };
+    };
+
+    filer = {
+      enable = lib.mkEnableOption "SeaweedFS filer server";
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 8888;
+        description = "Port for filer server.";
+      };
+
+      grpcPort = lib.mkOption {
+        type = lib.types.port;
+        default = 18888;
+        description = "gRPC port for filer server.";
+      };
+
+      ip = lib.mkOption {
+        type = lib.types.str;
+        description = "IP address to bind to.";
+      };
+
+      ipBind = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "IP address to bind to. If empty, defaults to same as ip.";
+      };
+
+      collection = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "All data will be stored in this default collection.";
+      };
+
+      dataDir = lib.mkOption {
+        type = lib.types.str;
+        default = "${baseDir}/filer";
+        description = "Data directory for filer.";
+      };
+
+      defaultReplicaPlacement = lib.mkOption {
+        type = lib.types.str;
+        default = "";
```

```suggestion
        default = cfg.<...>;
```
should we make this clear like that?